### ISSUE TITLE
glfwGetVersionString returning wrong version.

### ISSUE THREAD

**Issue by NicolayP** [glfw/glfw#1788]
`OS and version: Ubuntu 18.04`
`Release or commit: 0ef149c8f2451fbc8e866834675a075cfc295b6c`
`Error messages: None`

I was trying to get the version of the software just to check if my install is correct. 

Here is the code sample is used: 
```#include <GLFW/glfw3.h>
#include <iostream>

int main(){
    std::cout << GLFW_VERSION_MAJOR << "." << GLFW_VERSION_MINOR << "." << GLFW_VERSION_REVISION << std::endl;
    std::cout << glfwGetVersionString() << std::endl;
    return 0;
}
```

And here is the output produced on my machine:

```
3.4.0
3.2.0 X11 GLX EGL clock_gettime /dev/js Xf86vm shared
```

So i guess there is a bug somewhere.

**Comment by gracicot** [comment]
What are you compiler flag and how did you installed glfw?

**Comment by NicolayP** [comment]
This is the 3 instructions I ran from a build folder:
```
cmake -DBUILD_SHARED_LIBS=ON ..
make
sudo make install
```

**Comment by gracicot** [comment]
I presume you build your app using `-lglfw`? If that so, most likely you have the correct include flags but you link to the wrong library. In CMake, you'd have to do this:

    find_package(glfw3 3.4.0 REQUIRED) # You tell the version
    target_link_libraries(your_target PUBLIC glfw) # Link and add include dirs

If you link using the compiler directly, you have to replace `-lglfw` by `/usr/local/lib/libglfw3.a` or wherever you installed it.

**Comment by NicolayP** [comment]
So I just tried compiling with: 
`g++ glfw_test.cpp /usr/local/lib/libglfw3.so.3.4`
`g++ glfw_test.cpp /path/to/build_dir/libglfw3.so.3.4` just to make sure I'm using the right compiled lib.
and with a quick makefile using the `find_package` as metioned:
```
cmake_minimum_required(VERSION 3.5)

set(BIN_NAME test)
find_package(glfw3 3.4 REQUIRED)

add_executable(${BIN_NAME} glfw_test.cpp)

target_link_libraries(${BIN_NAME} PUBLIC glfw)
```

And 
`g++ glfw_test.cpp -lglfw`

and got the same result every time.

**Comment by NicolayP** [comment]
That was indeed the issue, I have Mujoco installed that comes shipped with a compiled version of glfw 3.2.0. 
Thank you for your help!
